src/pages/workspaces/workspace/analysis/modals/AnalysisModal.test.ts

```diff
@@ -0,0 +1,410 @@
+
+import { defaultAzureWorkspace, defaultGoogleWorkspace, galaxyDisk, galaxyRunning, getGoogleRuntime } from '../_testData/testData'
+import { AnalysisModal, AnalysisModalProps } from './AnalysisModal'
+
+
+const defaultGcpModalProps: AnalysisModalProps = {
+  isOpen: true,
+  workspace: defaultGoogleWorkspace,
+  location: 'US',
+  runtimes: [],
+  apps: [] as AppTool[],
+  appDataDisks: [],
+  persistentDisks: [],
+  onDismiss: () => {},
+  onError: () => {},
+  onSuccess: () => {},
+  openUploader: () => {},
+  uploadFiles: () => {},
+  //TODO: Temporary until Analyses.js implements useAnalysisFiles
+  refreshAnalyses: () => {}
+}
+
+const defaultAzureModalProps: AnalysisModalProps = {
+  ...defaultGcpModalProps,
+  workspace: defaultAzureWorkspace
+}
+
+jest.mock('src/libs/ajax/GoogleStorage')
+jest.mock('src/libs/ajax')
+
+jest.mock('src/libs/error', () => ({
+  ...jest.requireActual('src/libs/error'),
+  reportError: jest.fn(),
+}))
+
+jest.mock('src/libs/notifications', () => ({
+  notify: jest.fn((...args) => {
+    console.debug('######################### notify')/* eslint-disable-line */
+    console.debug({ method: 'notify', args: [...args] })/* eslint-disable-line */
+  })
+}))
+
+jest.mock('src/pages/workspaces/workspace/analysis/file-utils', () => {
+  const originalModule = jest.requireActual('src/pages/workspaces/workspace/analysis/file-utils')
+  return {
+    ...originalModule,
+    // getExtension: jest.fn().mockImplementation((ext: Extension) => originalModule.getExtension(ext)), //() => 'ipynb' as Extension,
+    useAnalysisFiles: jest.fn()
+  }
+})
+
+const createFunc = jest.fn()
+
+//TODO: Put in test data
+
+const getTestFile = (abs: AbsolutePath, cloudProvider: CloudProviderType = cloudProviderTypes.GCP): AnalysisFile => ({
+  name: abs,
+  ext: '.ipynb' as Extension,
+  displayName: getDisplayName(abs),
+  fileName: getFileName(abs),
+  tool: getToolLabelFromFileExtension(getExtension(abs)) as ToolLabel,
+  lastModified: new Date().getTime(),
+  cloudProvider
+})
+
+
+describe('AnalysisModal', () => {
+  beforeEach(() => {
+    // Arrange
+    asMockedFn(useAnalysisFiles).mockImplementation(() => ({
+      refresh: () => Promise.resolve(),
+      loadedState: { state: [], status: 'Ready' },
+      create: createFunc,
+      pendingCreate: { status: 'Ready', state: true }
+    }))
+
+    //@ts-expect-error
+    Ajax.mockImplementation(() => ({ Metrics: { captureEvent: jest.fn() } }))
+  })
+
+  it('GCP - Renders correctly by default', () => {
+    // Act
+    render(h(AnalysisModal, defaultGcpModalProps))
+    // Assert
+    screen.getByText('Select an application')
+    screen.getByAltText('Create new notebook')
+    screen.getByAltText('Create new R file')
+    screen.getByAltText('Create new Galaxy app')
+  })
+
+  it('GCP - Successfully resets view.', async () => {
+    // Arrange
+    render(h(AnalysisModal, defaultGcpModalProps))
+    const user = userEvent.setup()
```

Comment:
> We recommend invoking [userEvent.setup()](https://testing-library.com/docs/user-event/setup) before the component is rendered.

https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent

Reply:
Oops! I had a feeling the order was important. Fixing.